pyccel/parser/semantic.py

```diff
@@ -3015,12 +3005,6 @@ def _visit_FunctionTypeAnnotation(self, expr):
         res_type = self._visit(expr.result)
         return UnionTypeAnnotation(FunctionTypeAnnotation(arg_types, res_type))
 
-    def _visit_TypingFinal(self, expr):
-        annotation = self._visit(expr.arg)
-        for t in annotation:
-            t.is_const = True
-        return annotation
```

Comment:
Could you explain why you could remove this function?

Reply:
This was related to the code above. `TypingFinal` was visited as the base of the indexed element but needed special treatment to set `is_const` on the resulting `VariableTypeAnnotation`. This is no longer the case so it is sufficient to add the `_static_type` property to this class